game/scripts/vscripts/Sections/Chapter3/Chapter3.ts

```diff
@@ -0,0 +1,459 @@
+            if (isGoalStarted(key)) {
+                goals.push({ text: text, completed: isGoalCompleted(key) });
+            }
+        };
+
+        addGoal(NeutralGoalKeys.MoveToCamp, "Move to the neutral creep camp");
+        addGoal(
+            NeutralGoalKeys.UpgradeAbility,
+            "Upgrade the Dragon Blood ability"
+        );
+        addGoal(
+            NeutralGoalKeys.KillNeutralsSatyrs,
+            "Kill the neutral creeps (Satyrs)"
+        );
+        addGoal(NeutralGoalKeys.PickupItemArcaneRing, "Pickup the arcane ring");
+        addGoal(NeutralGoalKeys.UseNeutralItem, "Use your new found item.");
+        addGoal(NeutralGoalKeys.PressAlt, "Press alt to see the spawn box");
+        addGoal(
+            NeutralGoalKeys.MoveOutOfNeutralBox,
+            "Move out of the spawn box"
+        );
+        addGoal(
+            NeutralGoalKeys.KillNeutralsWolves,
+            "Kill the neutral creeps (Wolves)"
+        );
+        addGoal(NeutralGoalKeys.PickupItems, "Pick up the dropped items");
+        addGoal(
+            NeutralGoalKeys.SwitchItems,
+            "Switch the possessed mask with the arcane ring"
+        );
+        addGoal(NeutralGoalKeys.ShareItem, "Give the arcane ring to warlock");
+        addGoal(NeutralGoalKeys.ShareItem, "Put the item in the neutral stash");
+
+        return goals;
+    };
+
+    const playerHero = getPlayerHero();
+    if (!playerHero) error("Could not find the player's hero.");
+
+    let dragon_knight_dragon_blood = playerHero.FindAbilityByName(
+        "dragon_knight_dragon_blood"
+    );
+
+    if (!dragon_knight_dragon_blood) {
+        dragon_knight_dragon_blood = playerHero.AddAbility(
+            "dragon_knight_dragon_blood"
+        );
+    }
+
+    if (!dragon_knight_dragon_blood) {
+        error(
+            "dragon_knight_dragon_blood has been renamed or replaced by another ability, this is bad."
+        );
+    }
+
+    // Also in orderfilter at the bottom!
+    const giveAwayItemName = "item_arcane_ring";
+    const dropInStashItemName = "item_mysterious_hat";
+    const keepItemName = "item_possessed_mask";
+
+    const units: CDOTA_BaseNPC[] = [];
+
+    // if (IsInToolsMode()) {
+    //     playerHero.SetBaseDamageMin(1000);
+    //     playerHero.SetBaseDamageMax(1000);
+    // }
+
+    playerHero.SetMoveCapability(UnitMoveCapability.GROUND);
+    playerHero.SetAbsOrigin(GetGroundPosition(Vector(-4000, -550), undefined));
+
+    const markerLocation = Vector(-3250, -150);
+
+    const neutralCamp = Entities.FindByClassnameNearest(
+        "npc_dota_neutral_spawner",
+        Vector(-2608, -648, 265),
+        100
+    )!;
+    const neutralCampPostion = neutralCamp?.GetAbsOrigin();
+
+    GameRules.SpawnNeutralCreeps();
+    const neutralCamps = Entities.FindAllByClassname(
+        "npc_dota_neutral_spawner"
+    );
+
+    graph = tg.forkAny([
+        tg.trackGoals(getGoals),
+        tg.seq([
+            tg.wait(1),
```

Comment:
Might want to split this sequence up (can also be done after this is merged) into multiple functions returning a list of steps, ie 
```ts
tg.seq([
    ...goToCamp(),
    ...killSatyrCamp(),
   etc
```